Create dust when smashing furniture and terrain

ehughsbaird opened the pull request:
#### Summary
Features "Bashed furniture and terrain produce dust fields"

#### Purpose of change
Fixes https://github.com/CleverRaven/Cataclysm-DDA/issues/81556

#### Describe the solution
Allow bash info to specify fields added on hit and on destruction. Add a dust and splinters field to every furniture/terrain that bashes into `"splinter"`. 

See commits for more detail.

I also fixed a bug in the loading code for the `"bash"`/`"deconstruct"` fields for furniture/terrain.

#### Describe alternatives you've considered
I'm sure the field behavior could use improvement - I don't want it to fade away over time, but there are conditions that should clear it.

#### Testing
Zombies smashing up the conference room:
![zombies clustered around a table in a room filled with dust, with dust leaking out the windows and doors. There is only scattered dust in the rest of the building.](https://github.com/user-attachments/assets/8f509c8c-787f-43df-8077-09b3b4b7091b)
Setting off a grenade in an electronics store:
![There's a large explosion shaped area of dust, and no dust elsewhere](https://github.com/user-attachments/assets/33592f0a-b153-4fc0-8480-8150bd3b9304)
Aftermath of smashing up some benches with a stick:
![Where the three benches were and a little bit of the area north of them has dust.](https://github.com/user-attachments/assets/cdb49e34-0267-46a7-9a9d-a1b2f3d907fe)

PatrikLundell commented:
You should be able to clean it up. The quick-and-easy way would be to allow a mop to do it (like it does for blood and sludge), but a broom or similar would be more appropriate.

ehughsbaird committed:
Allow fields to choose to be moppable

Currently, all liquid fields are moppable. Allow fields that are not
liquid to select that they can also be cleaned with a mop.

ehughsbaird committed:
use was_loaded properly for bash and deconstruct

was_loaded indicates whether or not it is the first time this object has
been loaded.

The terrain/furniture loading code was passing the terrain/furniture
was_loaded value, when they could be creating the object and thus it
would be it's first time being loaded.

Pass whether or not the object has been created as the was_loaded value
instead of whether or not the terrain/furniture it's part of has.

ehughsbaird committed:
Allow bashed terrain/furniture to create fields

Allow specifying a field and an intensity to spawn that field at when an
object is hit, and when an object is destroyed. The field spawn when it
is hit will also spawn when it is destroyed.

ehughsbaird committed:
Add dust and splinters fields for bashed ter/furn

The dust is in the gas phase and can spread - this allows bashing to
spread dust over the place, instead of only in the tile bashed.

Similarly for splinters, but it's less likely to spread, and will try to
spread dust.

ehughsbaird committed:
Add dust to everything that splinters

All terrain/furniture that produce "splinter" in their bash results have
been given the dust and splinters field produced on bash.

ehughsbaird committed:
Mods furn/terrain smashes into dust

All terrain/furniture in mods that break into splinters produce the dust
and splinters field on bash.

ashGlaw commented:
This is neat but what is it's purpose?

I-am-Erk commented:
> This is neat but what is it's purpose?

It's cool.

The fields could be given relevant effects at high levels, and having furniture drop a field on bash is great infrastructure, but mostly it's just cool.
